NDTensors/ext/NDTensorsMetalExt/adapt.jl

```diff
@@ -6,7 +7,10 @@ end
 
 # More general than the version in Metal.jl
 function Adapt.adapt_storage(arraytype::Type{<:MtlArray}, xs::AbstractArray)
-  params = get_parameters(xs)
-  arraytype_specified = specify_parameters(arraytype, params...)
+  arraytype_specified = TypeParameterAccessors.specify_type_parameters(
+    arraytype,
+    (eltype, Base.ndims),
+    TypeParameterAccessors.type_parameters(xs, (eltype, Base.ndims)),
```

Comment:
@mtfishman I am not sure if this exactly is what you had in mind in with respect to this comment 
```
More general Adapt.adapt_storage definitions for GPU array types based on TypeParameterAccessors (see discussion in https://github.com/ITensor/ITensors.jl/pull/1350#discussion_r1517835891). This may involve defining a type_parameter_names function in TypeParameterAccessors as well as a less stringent version of type_parameters which outputs UnspecifiedTypeParameter if a parameter doesn’t exist, as opposed to erroring, which I think is the current behavior. It could be called something like maybe_unspecified_type_parameters.
```

Reply:
I am reverting these changes as we just discussed. Thanks